webapp/lib/view_model.dart

```diff
@@ -70,16 +74,19 @@ class MessageViewModel {
       checked = true;
     }
 
-    // Update the checkbox
-    codeSelectors.singleWhere((selector) => selector.scheme.id == schemeId).checked = checked;
-
     // Update the data-model by prepending this decision
     message.labels.insert(0,
       new Label(schemeId, new DateTime.now(), codeId,
         new Origin(auth.getUserEmail(), auth.getUserName()),
         checked: checked
         ));
     fbt.updateMessage(dataset, message);
+
+    // Update the checkbox and origin
+    CodeSelector codeSelector = codeSelectors.singleWhere((selector) => selector.scheme.id == schemeId);
+    codeSelector
+      ..checked = checked
+      ..origin = message.labels[0].labelOrigin.name;
```

Comment:
Can you factor the 'get the active label' (in this case done by labels[0]) from the rest? We may well need to be able to change this easily in the future and at the moment it's being duplicated in a couple of different places.

Reply:
Thanks for pointing this out. I could reuse the getLatestLabelForScheme method I created a while back instead of accessing the local most recent label. We're still duplicating getting the code selector for a scheme, but I'll send a PR following this one extracting that into a separate method.